### PR TITLE
Add @when_any and @when_not_all decorators

### PR DESCRIPTION
**Pull request by johnsca**

### CODE REVIEW

File: tests/test_helpers.py [hunk 1]
```diff
@@ -222,6 +222,29 @@ def test__when_all(self):
         self.kv.set('reactive.dispatch.phase', 'other')
         assert test(), 'when_all: other; both'
 
+    def test__when_any(self):
```
Comment (bcsaller) [inline tests/test_helpers.py]: Is there a reason some of the tests have double under test__ prefixes?

**Comment by bcsaller** [comment]
LGTM, thanks

File: tests/test_helpers.py (hunk 1 above)
Comment (johnsca) [inline tests/test_helpers.py]: The function being tested is `_when_any` but it's not strictly necessary.

**Comment by chuckbutler** [comment]
+1 LGTM -- however I have only run the tests, ymmv

**Pull request merged**